docs/proposals/csi.md

```diff
@@ -0,0 +1,349 @@
+      `node-driver-registrar` will use this unix domain socket to register the CSI driver’s unix domain socket with Edged.
+  * `plugin-dir` hostPath volume
+      Expose `/var/lib/edged/plugins/csi-nfs-driver/` from the edge node as `hostPath.type: DirectoryOrCreate`.
+      Mount inside `csi-nfs-driver` container which is the primary means of communication between Edged and the `csi-nfs-driver` container.
+  * `pods-mount-dir` hostPath volume
+      Expose `/var/lib/edged/pods` from the edge node.
+      Mount only in 'csi-nfs-driver' container at `/var/lib/edged/pods/`.
+      Ensure [bi-directional mount propagation](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) is enabled, 
+      so that any mounts setup inside this container are propagated back to edge node host machine.
+
+If it is not possible to deploy mutiple containers in  `DaemonSet` in some KubeEdge release versions,
+The cluster admins need to deploy two DaemonSets which includes `csi-nfs-driver` and `node-driver-registrar` respectively.
+
+### Example
+
+#### Static Provision Example
+
+Static Provision means the volume has already been created in somewhere.
+There is no need to create volume by the `External-Provisioner`.
+The customers can define it by PersistentVolume and PersistentVolumeClaim.
+Here list the CSI NFS Driver as an example.
+
+PersistentVolume Example:
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: csi-nfs-pv
+  labels:
+    name: csi-nfs-pv
+spec:
+  accessModes:
+  - ReadWriteMany
+  capacity:
+    storage: 1Gi
+  csi:
+    driver: csi-nfs-driver
+    volumeHandle: data-id
+    volumeAttributes:
+      server: 127.0.0.1
+      share: /export
+```
+Refer to the NFS share information and create a PersistentVolume.
+`server: 127.0.0.1` and `share: /export` in the `volumeAttributes` are used to speficy the NFS share information.
+
+PersistentVolumeClaim Example:
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: csi-nfs-pvc
+spec:
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 1Gi
+  selector:
+    matchExpressions:
+    - key: name
+      operator: In
+      values: [" csi-nfs-pv"]
+```
+Based on the PersistentVolume, this example is used to define PersistentVolumeClaim.
+Once the PersistentVolumeClaim is created successfully,
+the PersistentVolume will bound into PersistentVolumeClaim.
+
+Pod binding PersistentVolumeClaim Example:
+```yaml
+kind: Pod
+apiVersion: v1
+metadata:
+  name: my-csi-app
+spec:
+  containers:
+    - name: my-frontend
+      image: busybox
+      volumeMounts:
+      - mountPath: "/data"
+        name: my-csi-volume
+      command: [ "sleep", "1000000" ]
+  volumes:
+    - name: my-csi-volume
+      persistentVolumeClaim:
+        claimName: csi-nfs-pvc
+```
+`claimName: csi-hostpath-pvc` is the name of PersistentVolumeClaim.
+Before the pod is startup, volume will be attached into the edge node, and then mount into pod.
```

Comment:
Before the pod is startup, volume will be attached into the edge node, and then mount into pod.  -->
Before the pod starts up, the volume will be attached to the edge node, and then mounted inside the pod.

Reply:
Done